mobile/status_request_log.go

```diff
@@ -0,0 +1,88 @@
+package statusgo
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"runtime"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/logutils/requestlog"
+)
+
+var sensitiveRegex = regexp.MustCompile(`(?i)(".*?(password|mnemonic|openseaAPIKey|poktToken|alchemyArbitrumMainnetToken|raribleTestnetAPIKey|alchemyOptimismMainnetToken|statusProxyBlockchainUser|alchemyEthereumSepoliaToken|alchemyArbitrumSepoliaToken|infuraToken|raribleMainnetAPIKey|alchemyEthereumMainnetToken).*?")\s*:\s*("[^"]*")`)
+
+func getFunctionName(fn any) string {
+	return runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+}
+
+func getShortFunctionName(fn any) string {
+	fullName := getFunctionName(fn)
+	parts := strings.Split(fullName, ".")
+	return parts[len(parts)-1]
+}
+
+// logAndCall logs request call details and executes the fn function if logging is enabled
+func logAndCall(fn any, params ...any) any {
+	defer func() {
+		if r := recover(); r != nil {
+			// we're not sure if request logging is enabled here, so we log it use default logger
+			log.Error("panic found in logAndCall", "error", r, "stacktrace", string(debug.Stack()))
+			panic(r)
+		}
+	}()
+
+	var startTime time.Time
+
+	if requestlog.IsRequestLoggingEnabled() {
+		startTime = time.Now()
+	}
+
+	fnValue := reflect.ValueOf(fn)
+	fnType := fnValue.Type()
+	if fnType.Kind() != reflect.Func {
+		panic("fn must be a function")
+	}
+
```

Comment:
`logAndCall` expects `fn` to return at max 1 value. I am wondering whether we should add extra check for that:
```golang
	if fnType.NumOut() > 1 {
		panic("fn must return at max 1 value")
	}
```
or maybe be more flexible and return `[]any` instead.